src/middlewared/middlewared/plugins/pool.py

```diff
@@ -43,6 +43,9 @@
 
 RE_HISTORY_ZPOOL_SCRUB = re.compile(r'^([0-9\.\:\-]{19})\s+zpool scrub', re.MULTILINE)
 RE_HISTORY_ZPOOL_CREATE = re.compile(r'^([0-9\.\:\-]{19})\s+zpool create', re.MULTILINE)
+ZFS_CHECKSUM_CHOICES = [
+    'ON', 'OFF', 'FLETCHER2', 'FLETCHER4', 'SHA256', 'SHA512', 'SKEIN', 'EDONR',
```

Comment:
@amotin which of these should we leave?

Reply:
Is EDONR on FreeBSD?

Reply:
Are we sure we want to give the user the ability to turn off checksum? (Because the number one rule about dangerous options is that they will be picked up by a youtube influencer somewhere as a performance recommendation) :)

Reply:
EDONR is not available on FreeBSD.

Reply:
@freqlabs so what should be and what should not be present on FreeBSD and on Linux?

Reply:
I'm not entirely familiar with what's going on in this code. I'd probably present all the choices. But if an API consumer is able to enumerate the choices at run time, it might make sense to only offer the available choices on the platform, which is all of them on Linux, and all but edonr on FreeBSD. I don't think it's prudent to hide valid options.

Reply:
@anodos325 do you think it'll be fine if we keep `OFF` here but remove it from `pool.dataset.checksum_choices` so it won't be selectable in the UI?

Reply:
We cannot break on things that can be recklessly be done from a shell but we can prevent them from doing stupid things from UI, so remove the bad choices from the _choices API call seems reasonable.